lib/core/test/unit/Cardano/Wallet/Api/TypesSpec.hs

```diff
@@ -2157,6 +2159,11 @@ instance Arbitrary (PostTransactionOldData n) where
         <*> arbitrary
         <*> arbitrary
 
+instance Arbitrary TxMetadataWithSchema where
+  arbitrary = TxMetadataWithSchema
+    <$> elements [TxMetadataNoSchema, TxMetadataDetailedSchema]
```

Comment:
It might be worth considering the use of `arbitraryBoundedEnum` here, if you can derive `Bounded` and `Enum` for this type. Then we'd automatically get coverage of new constructors that are added. (If we add them.)

```suggestion
    <$> arbitraryBoundedEnum
```